solana/programs/example-native-token-transfers/src/instructions/luts.rs

```diff
@@ -0,0 +1,172 @@
+
+    /// CHECK: address lookup table program (checked by instruction)
+    #[account(executable)]
+    pub lut_program: AccountInfo<'info>,
+
+    pub system_program: Program<'info, System>,
+
+    /// These are the entries that will populate the LUT.
+    pub entries: Entries<'info>,
+}
+
+#[derive(Accounts)]
+pub struct Entries<'info> {
+    pub config: Account<'info, Config>,
+
+    #[account(
+        constraint = custody.key() == config.custody,
+    )]
+    pub custody: AccountInfo<'info>,
+
+    #[account(
+        constraint = token_program.key() == config.token_program,
+    )]
+    pub token_program: AccountInfo<'info>,
+
+    #[account(
+        constraint = mint.key() == config.mint,
+    )]
+    pub mint: AccountInfo<'info>,
+
+    #[account(
+        seeds = [crate::TOKEN_AUTHORITY_SEED],
+        bump,
+    )]
+    pub token_authority: AccountInfo<'info>,
+
+    pub outbox_rate_limit: Account<'info, OutboxRateLimit>,
+
+    // NOTE: this includes the system program so we don't need to add it in the outer context
+    pub wormhole: WormholeAccounts<'info>,
+}
+
+pub fn initialize_lut(ctx: Context<InitializeLUT>, recent_slot: u64) -> Result<()> {
+    let (ix, lut_address) = solana_address_lookup_table_program::instruction::create_lookup_table(
+        ctx.accounts.authority.key(),
+        ctx.accounts.payer.key(),
+        recent_slot,
+    );
+
+    // just a sanity check, should never be hit, so we don't provide a custom
+    // error message
+    assert_eq!(lut_address, ctx.accounts.lut_address.key());
+
+    // the LUT might already exist, in which case the new one will simply
+    // override it. Since we don't delete the old LUTs, this is safe -- clients
+    // holding references to old LUTs will still be able to use them.
+    ctx.accounts.lut.set_inner(LUT {
+        bump: ctx.bumps.lut,
+        address: lut_address,
+    });
+
+    // NOTE: LUTs can be permissionlessly created (i.e. the authority does
+    // not need to sign the transaction). This means that the LUT might
+    // already exist (if someone frontran us). However, it's not a problem:
+    // AddressLookupTable::create_lookup_table checks if the LUT already
+    // exists and does nothing if it does.
+    //
+    // LUTs can only be created permissionlessly, but only the authority is
+    // authorised to actually populate the fields, so we don't have to worry
+    // about the frontrunner populating it with junk. The only risk of that would
+    // be the LUT being filled to capacity (256 addresses), with no
+    // possibility for us to add our own accounts -- no other security impact.
+    invoke(
+        &ix,
+        &[
+            ctx.accounts.lut_address.to_account_info(),
+            ctx.accounts.authority.to_account_info(),
+            ctx.accounts.payer.to_account_info(),
+            ctx.accounts.system_program.to_account_info(),
+        ],
+    )?;
+
+    let entries_infos = ctx.accounts.entries.to_account_infos();
```

Comment:
Would there be any benefit (maybe CU savings?) to failing early on this line? The program could check whether entries_info.len() + 1  exceeds 256 (max LUT size) instead of doing the CPI. We know that the CPI will fail in this case so we may as well not call itIf so, it could even be moved above the first invoke.

The only downside that comes to mind is wiring the LUT max size constant into the program, but maybe this could be imported from the Solana LUT crate. I imagine it's not likely to change either, but not sure